### ISSUE TITLE
Cannot build from source on RPi

### ISSUE THREAD

**Issue by zaneclaes** [badaix/snapcast#902]
**Describe the bug**
I want to compile a snapserver from source so that I may decrease the hard-coded buffer floor (per https://github.com/badaix/snapcast/issues/329#issuecomment-899822450).

When I follow the instructions to do so, I seem to have problems with the boost headers. I've downloaded v1.77.0 and unzipped it, and executed the following build command: `ADD_CFLAGS="-I/home/pi/boost_1_77_0/" make`

```
make -C client
make[1]: Entering directory '/home/pi/snapcast/client'
g++ -O2 -I/home/pi/boost_1_77_0/ -std=c++14 -Wall -Wextra -Wpedantic -Wno-unused-function -DBOOST_ERROR_CODE_HEADER_ONLY -DHAS_FLAC -DHAS_OGG -DHAS_OPUS -DHAS_SOXR -DVERSION=\"0.25.0\" -I. -I.. -I../common -pthread -DHAS_VORBIS -DHAS_ALSA -DHAS_PULSE -DHAS_AVAHI -DHAS_DAEMON -c player/player.cpp -o player/player.o
In file included from /home/pi/boost_1_77_0/boost/process/detail/basic_cmd.hpp:14,
                 from /home/pi/boost_1_77_0/boost/process/args.hpp:33,
                 from player/player.cpp:29:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<wchar_t> > boost::process::detail::char_converter<wchar_t, std::vector<std::__cxx11::basic_string<char> > >::conv(const std::vector<std::__cxx11::basic_string<char> >&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:150:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<wchar_t> > boost::process::detail::char_converter<wchar_t, std::initializer_list<std::__cxx11::basic_string<char> > >::conv(const std::initializer_list<std::__cxx11::basic_string<char> >&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:165:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<wchar_t> > boost::process::detail::char_converter<wchar_t, std::vector<char*> >::conv(const std::vector<char*>&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:180:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<wchar_t> > boost::process::detail::char_converter<wchar_t, std::initializer_list<char*> >::conv(const std::initializer_list<char*>&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:197:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<char> > boost::process::detail::char_converter<char, std::vector<std::__cxx11::basic_string<wchar_t> > >::conv(const std::vector<std::__cxx11::basic_string<wchar_t> >&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:214:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<char> > boost::process::detail::char_converter<char, std::initializer_list<std::__cxx11::basic_string<wchar_t> > >::conv(const std::initializer_list<std::__cxx11::basic_string<wchar_t> >&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:229:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<char> > boost::process::detail::char_converter<char, std::vector<wchar_t*> >::conv(const std::vector<wchar_t*>&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:244:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp: In static member function ‘static std::vector<std::__cxx11::basic_string<char> > boost::process::detail::char_converter<char, std::initializer_list<wchar_t*> >::conv(const std::initializer_list<wchar_t*>&)’:
/home/pi/boost_1_77_0/boost/process/detail/traits/wchar_t.hpp:261:14: error: ‘transform’ is not a member of ‘std’
         std::transform(in.begin(), in.end(), ret.begin(),
              ^~~~~~~~~
make[1]: *** [Makefile:123: player/player.o] Error 1
make[1]: Leaving directory '/home/pi/snapcast/client'
```

**Environment details**
- OS: Raspbian
- Snapcast version: compiled from source

**Comment by badaix** [comment]
Seems like boost::process requires C++17 in boost 1.77 :(
You can either use Boost 1.76 or replace `-std=c++14` with `-std=c++17` in `client/Makefile` and `server/Makefile`

**Comment by zaneclaes** [comment]
Thanks! That worked.

FWIW, reducing the buffer size was also successful. I got down to ~120ms before noticing any distortion. Using a USB mixer as an alsa input source, I can now connect instruments directly into Snapcast to play over all 30+ speakers in every room in the house with low enough latency that the people playing the music are able to cope ;)

**Comment by kingosticks** [comment]
Purely out of interest, what does snapcast offer that the programs dedicated to this use case (super low latency streaming) don't? Programs like JamRTC, Sonobus, Jamulus etc?

**Comment by zaneclaes** [comment]
@kingosticks I really would like to research those more, but here's my use-case. I started with snapcast as a way to airplay to all the speakers in the house via the `shairport-sync` integration. I integrated it into my Home Assistant IOT deployment, so that we can control each zone independently. This covers 99% of our use — but next week we're hosting an event at our house, and I hoped to be able to also play music with other musicians who will be attending (I'm still pretty new to the piano/keyboard, but friends play guitar, flute, etc.) so it seemed like it'd be fun to be able to play the music to all the different speaker zones. I'd definitely be interested in a lower latency solution for this part, but I would need to figure out how it fit in alongside Home Assistant and snapcast airplay (open to suggestions!)

<img width="507" alt="Screen Shot 2021-08-21 at 10 47 02 AM" src="https://user-images.githubusercontent.com/2084821/130329249-7c6ec6f6-e66c-4a93-a6f4-b9ad5cb8eadf.png">
